Source/SwiftLintFramework/Extensions/Configuration+Merging.swift

```diff
@@ -10,27 +10,26 @@ extension Configuration {
     }
 
     private func configuration(forPath path: String) -> Configuration {
-        if path == rootDirectory {
-            return self
-        }
-
         let pathNSString = path.bridge()
         let configurationSearchPath = pathNSString.appendingPathComponent(Configuration.fileName)
 
+        if configurationSearchPath == configurationPath {
+            return self
+        }
+
         // If a configuration exists and it isn't us, load and merge the configurations
         if configurationSearchPath != configurationPath &&
             FileManager.default.fileExists(atPath: configurationSearchPath) {
             let fullPath = pathNSString.absolutePathRepresentation()
             let customRuleIdentifiers = (rules.first(where: { $0 is CustomRules }) as? CustomRules)?
                 .configuration.customRuleConfigurations.map { $0.identifier }
-            let config = Configuration.getCached(atPath: fullPath) ??
```

Comment:
I removed `Configuration.getCached(atPath: fullPath)` because it is already done in init that we are calling from here. Difference is that `getCached` in that init has extra functionality that is missing from here.

https://github.com/realm/SwiftLint/blob/3a36212/Source/SwiftLintFramework/Models/Configuration.swift#L141